From 32d29aa5cbf7888acd0d4537181030df0a9f3ff3 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Mon, 21 Nov 2022 10:21:32 +0530
Subject: [PATCH] Display is not alerted for message in sleep mode.

Wake Screen for notifications feature was not enabled,
So it was not waking up the screen.

Enabled this feature and can be enable/disble from
Settings->Display->Lock screen option.

Tracked-On: OAM-104595
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 core/res/res/values/config.xml                                 | 2 +-
 packages/SettingsProvider/res/values/defaults.xml              | 3 +++
 .../src/com/android/providers/settings/DatabaseHelper.java     | 3 +++
 3 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 6995cc3b94ea..7c28baed61e8 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -2439,7 +2439,7 @@
 
          Note that doze dreams are not subject to the same start conditions as ordinary dreams.
          Doze dreams will run whenever the power manager is in a dozing state. -->
-    <string name="config_dozeComponent" translatable="false"></string>
+    <string name="config_dozeComponent" translatable="false">com.android.systemui/com.android.systemui.doze.DozeService</string>
 
     <!-- If true, the doze component is not started until after the screen has been
          turned off and the screen off animation has been performed. -->
diff --git a/packages/SettingsProvider/res/values/defaults.xml b/packages/SettingsProvider/res/values/defaults.xml
index 7732da40aac2..a827c99ed73f 100644
--- a/packages/SettingsProvider/res/values/defaults.xml
+++ b/packages/SettingsProvider/res/values/defaults.xml
@@ -177,6 +177,9 @@
     <!-- Default for Settings.Secure.WAKE_GESTURE_ENABLED -->
     <bool name="def_wake_gesture_enabled">true</bool>
 
+    <!-- Default for Settings.Secure.DOZE_ENABLED -->
+    <bool name="def_doze_enabled">true</bool>
+
     <!-- Default state of tap to wake -->
     <bool name="def_double_tap_to_wake">true</bool>
 
diff --git a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
index b851232ace82..a92f11a1fb54 100644
--- a/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
+++ b/packages/SettingsProvider/src/com/android/providers/settings/DatabaseHelper.java
@@ -2381,6 +2381,9 @@ class DatabaseHelper extends SQLiteOpenHelper {
             loadBooleanSetting(stmt, Settings.Secure.WAKE_GESTURE_ENABLED,
                     R.bool.def_wake_gesture_enabled);
 
+            loadBooleanSetting(stmt, Settings.Secure.DOZE_ENABLED,
+                    R.bool.def_doze_enabled);
+
             loadIntegerSetting(stmt, Secure.LOCK_SCREEN_SHOW_NOTIFICATIONS,
                     R.integer.def_lock_screen_show_notifications);
 
-- 
2.17.1

